lib/Infer/EnumerativeSynthesis.cpp

```diff
@@ -216,27 +226,30 @@ void getGuesses(std::vector<Inst *> &Guesses,
 
     for (auto I = Comps.begin(); I != Comps.end(); ++I) {
       // Prune: only one of (mul x, C), (mul C, x) is allowed
-      if (Inst::isCommutative(K) && (*I)->K == Inst::ReservedConst)
+      if ((Inst::isCommutative(K) || Inst::isOverflowIntrinsicMain(K) || Inst::isOverflowIntrinsicSub(K)) &&
+          (*I)->K == Inst::ReservedConst)
         continue;
 
       // Prune: I1 should only be the first argument
       if ((*I)->K == Inst::ReservedInst && (*I) != I1)
         continue;
 
       // PRUNE: don't try commutative operators both ways
-      auto Start = Inst::isCommutative(K) ? I : Comps.begin();
+      auto Start = (Inst::isCommutative(K) || Inst::isOverflowIntrinsicMain(K) || Inst::isOverflowIntrinsicSub(K)) ? I : Comps.begin();
```

Comment:
why not just make these intrinsics commutative instead of adding a special case here?

Reply:
also, avoid lines >80 columns

Reply:
Because sadd.with.overflow is actually not commutative in internal souper representation. It's first argument is basic `add` operation and second argument is one among those AddO variants. However, it is commutative in enumerative synthesis algorithm and on a souper IR level. Hence, this special case.